cpp/tests/groupby/group_max_test.cpp

```diff
@@ -196,5 +196,77 @@ TEST_F(groupby_dictionary_max_test, basic)
                   force_use_sort_impl::YES);
 }
 
+template <typename T>
+struct FixedPointTestBothReps : public cudf::test::BaseFixture {
+};
+
+TYPED_TEST_CASE(FixedPointTestBothReps, cudf::test::FixedPointTypes);
+
+TYPED_TEST(FixedPointTestBothReps, GroupBySortMaxDecimalAsValue)
+{
+  using namespace numeric;
+  using decimalXX  = TypeParam;
+  using RepType    = cudf::device_storage_type_t<decimalXX>;
+  using fp_wrapper = cudf::test::fixed_point_column_wrapper<RepType>;
+  using K          = int32_t;
+
+  for (auto const i : {2, 1, 0, -1, -2}) {
+    auto const scale = scale_type{i};
+    auto const keys  = fixed_width_column_wrapper<K>{1, 2, 3, 1, 2, 2, 1, 3, 3, 2};
+    auto const vals  = fp_wrapper{{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}, scale};
+
+    auto const expect_keys     = fixed_width_column_wrapper<K>{1, 2, 3};
+    auto const expect_vals_max = fp_wrapper{{6, 9, 8}, scale};
+
+    auto agg3 = cudf::make_max_aggregation();
+    test_single_agg(
+      keys, vals, expect_keys, expect_vals_max, std::move(agg3), force_use_sort_impl::YES);
+  }
+}
+
+// This test will not work until the following ptxas bug is fixed in 10.2
+// https://nvbugswb.nvidia.com/NvBugs5/SWBug.aspx?bugid=3186317&cp=
+TYPED_TEST(FixedPointTestBothReps, DISABLED_GroupByHashMaxDecimalAsValue)
+{
+  using namespace numeric;
+  using decimalXX  = TypeParam;
+  using RepType    = cudf::device_storage_type_t<decimalXX>;
+  using fp_wrapper = cudf::test::fixed_point_column_wrapper<RepType>;
+  using K          = int32_t;
+
+  for (auto const i : {2, 1, 0, -1, -2}) {
+    auto const scale = scale_type{i};
+    auto const keys  = fixed_width_column_wrapper<K>{1, 2, 3, 1, 2, 2, 1, 3, 3, 2};
+    auto const vals  = fp_wrapper{{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}, scale};
+
+    auto const expect_keys     = fixed_width_column_wrapper<K>{1, 2, 3};
+    auto const expect_vals_max = fp_wrapper{{6, 9, 8}, scale};
+
+    auto agg7 = cudf::make_max_aggregation();
+    test_single_agg(keys, vals, expect_keys, expect_vals_max, std::move(agg7));
+  }
+}
+
+TYPED_TEST(FixedPointTestBothReps, GroupBySortMaxDecimalAsValueAndKey)
```

Comment:
You don't need to add a test that check decimal as key in this file. Since the grouping of keys and aggregation of values are separate, they were put in separate test files. As long as there's a test for decimal as keys in `groupby_keys_test.cpp` in this line then there's sufficient coverage. https://github.com/rapidsai/cudf/blob/6828e2ce559569c55b8f04e660a93ab4c21f25b4/cpp/tests/groupby/groupby_keys_test.cpp#L31

Reply:
So you want to remove the test?

Reply:
And add decimal to the other keys test

Reply:
Done.